src/pal/src/file/directory.cpp

```diff
@@ -383,21 +377,20 @@ GetCurrentDirectoryW(
              OUT LPWSTR lpBuffer)
 {
     DWORD dwWideLen = 0;
-    DWORD dwLastError = 0;
-
-    char  *current_dir;
+    DWORD dwLastError = ERROR_BAD_PATHNAME;
     int   dir_len;
+    char  *current_dir;
 
     PERF_ENTRY(GetCurrentDirectoryW);
     ENTRY("GetCurrentDirectoryW(nBufferLength=%u, lpBuffer=%p)\n",
           nBufferLength, lpBuffer);
 
-    current_dir = PAL__getcwd( NULL, MAX_LONGPATH + 1 );
 
+    current_dir = PAL__getcwd( NULL, 0 );
+    
     if ( !current_dir )
     {
         WARN( "PAL__getcwd returned NULL\n" );
```

Comment:
Why don't we get the last error from the errno anymore? There can be multiple different kinds of error - permissions problem, name too long, OOM, ...